gobblin-core/src/main/java/gobblin/writer/test/TestingEventBusAsserter.java

```diff
@@ -0,0 +1,169 @@
+        Function<TestingEventBuses.Event, String> messageGen
+        ) throws InterruptedException, TimeoutException {
+    TestingEventBuses.Event nextEvent = _events.pollFirst(_defaultTimeoutValue, _defaultTimeoutUnit);
+    if (null == nextEvent) {
+      throw new TimeoutException();
+    }
+    if (!predicate.apply(nextEvent)) {
+      throw new AssertionError(messageGen.apply(nextEvent));
+    }
+    return nextEvent;
+  }
+
+  /**
+   * Variation on {@link #assertNext(Predicate, Function)} with a constant message.
+   */
+  public TestingEventBuses.Event assertNext(final Predicate<TestingEventBuses.Event> predicate,
+        final String message) throws InterruptedException, TimeoutException {
+    return assertNext(predicate, new StaticMessage(message));
+  }
+
+
+  /** Similar to {@link #assertNext(Predicate, Function)} but predicate is on the value directly. */
+  public <T> TestingEventBuses.Event assertNextValue(final Predicate<T> predicate ,
+        Function<TestingEventBuses.Event, String> messageGen)
+            throws InterruptedException, TimeoutException {
+    return assertNext(new Predicate<TestingEventBuses.Event>() {
+      @Override public boolean apply(@Nonnull Event input) {
+        return predicate.apply(input.<T>getTypedValue());
+      }
+    }, messageGen);
+  }
+
+  /** Similar to {@link #assertNext(Predicate, String)} but predicate is on the value directly. */
+  public <T> TestingEventBuses.Event assertNextValue(final Predicate<T> predicate, String message)
+         throws InterruptedException, TimeoutException {
+    return assertNext(new Predicate<TestingEventBuses.Event>() {
+      @Override public boolean apply(@Nonnull Event input) {
+        return predicate.apply(input.<T>getTypedValue());
+      }
+    }, message);
+  }
+
+  public <T> TestingEventBuses.Event assertNextValueEq(final T expected)
+         throws InterruptedException, TimeoutException {
+    return assertNextValue(Predicates.equalTo(expected),
+        new Function<TestingEventBuses.Event, String>() {
+          @Override public String apply(@Nonnull Event input) {
+            return "Event value mismatch: " + input.getValue() + " != " + expected;
+          }
+    });
+  }
+
+  /**
+   * Verify that all next several values are a permutation of the expected collection of event
+   * values. This method allows for testing that certain values are produced in some random order.
+   * Blocking assert.  */
+  public <T> void assertNextValuesEq(final Collection<T> expected)
+         throws InterruptedException, TimeoutException {
+    final Set<T> remainingExpectedValues = new HashSet<>(expected);
+    final Predicate<T> checkInRemainingAndRemove = new Predicate<T>() {
+      @Override public boolean apply(@Nonnull T input) {
+        if (! remainingExpectedValues.contains(input)) {
+          return false;
+        }
+        remainingExpectedValues.remove(input);
```

Comment:
This specific asset implementation will fail. It is meant to check only for permutations of distinct values.